spring-integration-file/src/main/java/org/springframework/integration/file/filters/FileListFilter.java

```diff
@@ -46,16 +45,15 @@
 	 * @since 5.2
 	 */
 	default boolean accept(F file) {
-		@SuppressWarnings("unchecked")
-		F[] fileArray = (F[]) Array.newInstance(file.getClass(), 1);
-		fileArray[0] = file;
-		return filterFiles(fileArray).size() > 0;
+		throw new UnsupportedOperationException(
+				"Filters that return true in supportsSingleFileFiltering() must implement this method");
 	}
 
 	/**
 	 * Indicates that this filter supports filtering a single file.
 	 * Default false.
 	 * @return true to allow external calls to {@link #accept(Object)}.
+	 * @since 5.2
 	 */
 	default boolean supportsSingleFileFiltering() {
```

Comment:
Can we also mention in JavaDoc for this method that `accept()` must be implemented in case of `true` ?

do you mention that in Docs though?

Thanks